src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -554,27 +554,289 @@ internal static int NonWindowsGetProcessParentPid(int pid)
+                public DateTime AccessTime;
+
+                /// <summary>The last modified time for the filesystem item.</summary>
+                public DateTime ModifiedTime;
+
+                /// <summary>The last time the status changes for the filesystem item.</summary>
+                public DateTime StatusChangeTime;
+
+                /// <summary>The block size of the filesystem.</summary>
+                public long BlockSize;
+
+                /// <summary>The device id of the filesystem item.</summary>
+                public int DeviceId;
+
+                /// <summary>The number of blocks used by the filesystem item.</summary>
+                public int NumberOfBlocks;
+
+                /// <summary>The type of the filesystem item.</summary>
+                public ItemType ItemType;
+
+                /// <summary>Whether the filesystem item has the setuid bit enabled.</summary>
+                public bool IsSetUid;
+
+                /// <summary>Whether the filesystem item has the setgid bit enabled.</summary>
+                public bool IsSetGid;
+
+                /// <summary>Whether the filesystem item has the sticky bit enabled. This is only available for directories.</summary>
+                public bool IsSticky;
+
+                // helper for getting unix mode
+                private Dictionary<StatMask, string> modeMap = new Dictionary<StatMask, string>()
+                {
+                        { StatMask.OwnerRead, "r" },
+                        { StatMask.OwnerWrite, "w" },
+                        { StatMask.OwnerExecute, "x" },
+                        { StatMask.GroupRead, "r" },
+                        { StatMask.GroupWrite, "w" },
+                        { StatMask.GroupExecute, "x" },
+                        { StatMask.OtherRead, "r" },
+                        { StatMask.OtherWrite, "w" },
+                        { StatMask.OtherExecute, "x" },
+                };
+
+                /// <summary>Convert the mode to a string which is usable in our formatting.</summary>
+                /// <returns>The mode converted into a Unix style string similar to the output of ls.</returns>
+                public string GetModeString()
+                {
+                    StatMask[] permissions = new StatMask[]
+                    {
+                        StatMask.OwnerRead,
+                        StatMask.OwnerWrite,
+                        StatMask.OwnerExecute,
+                        StatMask.GroupRead,
+                        StatMask.GroupWrite,
+                        StatMask.GroupExecute,
+                        StatMask.OtherRead,
+                        StatMask.OtherWrite,
+                        StatMask.OtherExecute
+                    };
+
+                    // start the mode string with the ItemType
+                    System.Text.StringBuilder sb = new System.Text.StringBuilder();
```

Comment:
The length of the output string is fixed to 10 i guess, so we should initialize the string builder with 10 as the capacity.

Reply:
i rewrote this method using a char array.